src/apispec/ext/marshmallow/common.py

```diff
@@ -95,7 +95,7 @@ def filter_excluded_fields(fields, Meta, exclude_dump_only):
     :param Meta: the schema's Meta class
     :param bool exclude_dump_only: whether to filter fields in Meta.dump_only
     """
-    exclude = getattr(Meta, "exclude", [])
+    exclude = list(getattr(Meta, "exclude", []))
     if exclude_dump_only:
         exclude += getattr(Meta, "dump_only", [])
```

Comment:
I think `dump_only` could be a tuple as well.

Reply:
Yes. I tested this a little more:
tuple + tuple is OK
list + list is also OK
That's why it only throws error, when one of them is set to a tuple and the other receives its default value as list.

So it should always be the same type before concatenation, not necessarily list, could be tuple.
The list's extend method might result in a more elegant solution.